src/main/java/com/desiato/puresynth/services/SessionService.java

```diff
@@ -6,22 +6,36 @@
 import com.desiato.puresynth.repositories.SessionRepository;
 import com.desiato.puresynth.repositories.UserRepository;
 import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Service;
 
 import java.util.Optional;
 import java.util.UUID;
 
 @AllArgsConstructor
+@Slf4j
 @Service
 public class SessionService {
 
     private final SessionRepository sessionRepository;
     private final UserRepository userRepository;
 
     public Session createSession(User user) {
+        if (user.getEmail() == null) {
+            throw new IllegalStateException("User email must not be null");
+        }
         String tokenValue = UUID.randomUUID().toString();
         Session newSession = new Session(tokenValue, user.getId());
-        sessionRepository.save(newSession);
+        log.info("Token to be assigned: {}", tokenValue);
+        log.info("User ID to be assigned: {}", user.getId());
+        try {
```

Comment:
What's point of this?

Reply:
It helped me better isolate the fact that the `sessionService.save()` method was the problematic point.

Reply:
And now, it should be reverted. There is no justification, why it should stay in this state.